src/app.js

```diff
@@ -5,13 +5,33 @@ import cors from 'cors';
 import logger from 'morgan';
 import passport from 'passport';
 import dotenv from 'dotenv';
+import socketIO from 'socket.io';
 import session from 'express-session';
 import './utilities/socialAuthStrategies';
 import routes from './routes';
 
 // Create global app object
 const app = express();
 
+// let's start our server...
+const server = app.listen(process.env.PORT || 4001, () => {
+  console.log(`Server is up and running on port ${server.address().port}...!`); //eslint-disable-line
+});
+
+const io = socketIO(server);
+
+io.on('connection', (socket) => {
+  socket.on('inAppNotifications', (data) => {
+    socket.emit('inAppNotifications', data);
+  });
+});
+
+const ioMiddleware = (req, res, next) => {
+  req.io = io;
+  next();
+};
+
```

Comment:
I think this middleware definition, from line 29, should be in the middleware folder and not in this file so that the application can be modular, easily maintainable, and easy to test.

Reply:
Using the middleware index file will give the middleware access to all the routes and it means I don't have to include them in specific routes. It's the way Express works.